src/drivers/mediatek/mt8195/timer.c

```diff
@@ -59,14 +57,8 @@ int64_t platform_timer_set(struct timer *timer, uint64_t ticks)
 
 	/*ostimer 13M counter to 26M interrupt */
 	time = (((uint64_t)high << 32) | low) << 1;
-
-	if (ticks > time)
-		ticks_set = (uint32_t)(ticks - time);
-	else
-		ticks_set = 1; /*set 1 to trigger interrupt*/
-
-	lowtimeout = ticks_set;
-	timer->hitimeout = hitimeout;
+	ticks_set = (ticks > time) ? ticks - time : 1;
```

Comment:
@kuanhsuncheng since these are both unsigned and we are checking for overflow, would this be better instead of setting overflow to 1
```c
ticks_set  (ticks > time) ? ticks - time : MAX_UINT64 - time  ticks;
```
This would try and keep the desired period at overflow.

Reply:
ok I think that 's great.
thanks.